src/Data/Avro/Codec.hs

```diff
@@ -0,0 +1,116 @@
+{-# LANGUAGE BangPatterns        #-}
+{-# LANGUAGE OverloadedStrings   #-}
+{-# LANGUAGE RankNTypes          #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+module Data.Avro.Codec (
+    Codec(..)
+  , Decompress
+  , nullCodec
+  , deflateCodec
+  ) where
+
+import           Codec.Compression.Zlib.Internal as Zlib
+import qualified Data.Binary.Get                 as G
+import           Data.ByteString                 (ByteString)
+import qualified Data.ByteString                 as BS
+import qualified Data.ByteString.Lazy            as LBS
+
+-- | Block decompression function for blocks of Avro.
+type Decompress a = LBS.ByteString -> G.Get a -> Either String a
+
+-- | A `Codec` allows for compression/decompression of a block in an
+-- Avro container according to the Avro spec.
+data Codec = Codec
+  {
+    -- | The name of the codec according to the Avro spec.
+    codecName       :: ByteString
```

Comment:
Maybe we should make this an enum? The spec only specifies three possible codecs: null, deflate and (optionally) snappy:

```
data CodecType = Null | Optional | Snappy deriving (..)
```

Reply:
I am not sure it's worth it at this point. It's all very encapsulated already.